capstone/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,61 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import java.io.BufferedReader;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.data.PerspectiveInput;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/call-perspective")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Get the input
+    String data = request.getReader().readLine();
+    Gson gson = new Gson();		
+    PerspectiveInput info = gson.fromJson(data, PerspectiveInput.class);  
+    
+    String text = info.getText();
+    String lang = info.getLang();
+		
```

Comment:
Just something to note for later - we might want to add error handling for blank inputs here

Reply:
Got it. For language, we should have an empty string if nothing is entered by the user, as Perspective will guess in that case.

Reply:
Perspective sometimes fail to detect a language and returns an error, so we'll eventually want to have some error handling here to handle that.